docs/configuration.rst

```diff
@@ -0,0 +1,233 @@
+    $configuration = new Configuration(['server' => 'http://public@example.com/1']);
+
+By default this option is borrowed from the ``SENTRY_DSN`` environment variable,
+if it exists. A value of ``null`` disables completely the sending of any event.
+Since Sentry 1.9 the public DSN must be used instead of the private one, which
+is deprecated and whose support will be removed in future releases of the server.
+
+Server name
+===========
+
+The name of the server the SDK is running on (it's usually the hostname).
+
+.. code-block:: php
+
+    $configuration = new Configuration(['server_name' => 'foo']);
+    $configuration->setServerName('foo');
+
+By default this option is set to the hostname of the server the SDK is running
+on retrieved from a call to the ``gethostname()`` method.
+
+Release
+=======
+
+The release tag to be passed with every event sent to Sentry. This permits to
+track in which versions of your application each error happens.
+
+.. code-block:: php
+
+    $configuration = new Configuration(['release' => '1.2.3']);
+    $configuration->setRelease('1.2.3');
+
+Logger
+======
+
+The name of the logger which creates the events.
+
+.. code-block:: php
+
+    $configuration = new Configuration(['logger' => 'foo']);
+    $configuration->setLogger('foo');
+
+By default this option is set to ``php``.
+
+Excluded loggers
+================
+
+The list of logger 'progname's to exclude from breadcrumbs.
+
+.. code-block:: php
+
+    $configuration = new Configuration(['excluded_loggers' => ['foo']);
+    $configuration->setExcludedLoggers(['foo']);
+
+Project root
+============
+
+The root of the project source code. As Sentry is able to distinguish project
+files from third-parties ones (e.g. vendors), this option can be configured to
+mark the directory containing all the source code of the application.
+
+.. code-block:: php
+
+    $configuration = new Configuration(['project_root' => '/foo/bar']);
+    $configuration->setProjectRoot('/foo/bar');
+
+For example, assuming that the directory structure shown below exists, marking
+the project root as ``project-folder/src/`` means that every file inside that
+directory that is part of a stacktrace frame will be marked as "application
+code".
+
+.. code-block::
+
+    project-folder/
+    ├── vendor/
+        ├── foo/
+    ├── src/
+        ├── bar/ <-- these are going to be marked as application files
+
+Current environment
+===================
+
+The name of the current environment. There can be multiple environments per
+application, and each event belongs to one of them.
+
+.. code-block:: php
+
+    $configuration = new Configuration(['current_environment' => 'foo']);
+    $configuration->setCurrentEnvironment('foo');
+
+Environments
+============
+
+The environments are a feature that allows events to be easily filtered in
+Sentry. An application can have multiple environments, but just one is active
+at the same time. This option let you configure the environments names: if
```

Comment:
What's the default behavior here? What's the default value?

Reply:
I've added a description of the default behavior and of its default value